### ISSUE TITLE
Importing NumPy to R using reticulate package failed

### ISSUE THREAD

**Issue by marinaesp** [numpy/numpy#17518]
Hey,

I'm new to python and just trying to import `NumPy` using `reticulate`, but failed. I guess it has something to do with my configuration of python, but I can't solve it.  Here is the steps with outputs:

### Steps
``` r
library(reticulate)
use_condaenv( "r-reticulate")
reticulate::use_python("/Users/marinaespinasse/Library/r-miniconda/envs/r-reticulate/bin/python", required = TRUE)
reticulate::py_config()
```
python:         /Users/marinaespinasse/Library/r-miniconda/envs/r-reticulate/bin/python
libpython:      /Users/marinaespinasse/Library/r-miniconda/envs/r-reticulate/lib/libpython3.6m.dylib
pythonhome:     /Users/marinaespinasse/Library/r-miniconda/envs/r-reticulate:/Users/marinaespinasse/Library/r-miniconda/envs/r-reticulate
version:        3.6.9 |Intel Corporation| (default, Sep 11 2019, 11:39:53)  [GCC 4.2.1 Compatible Apple LLVM 7.3.0 (clang-703.0.31)]
numpy:          /Users/marinaespinasse/Library/r-miniconda/envs/r-reticulate/lib/python3.6/site-packages/numpy
numpy_version:  1.17.0
numpy:          /Users/marinaespinasse/Library/r-miniconda/envs/r-reticulate/lib/python3.6/site-packages/numpy

```
numpy <- import("numpy")
```

### Error message
Importing the numpy c-extensions failed.
- Try uninstalling and reinstalling numpy.
- If you have already done that, then:
  1. Check that you expected to use Python3.6 from "/Users/marinaespinasse/Library/r-miniconda/envs/r-reticulate/bin/python",
     and that you have no directories in your PATH or PYTHONPATH that can
     interfere with the Python and numpy version "1.17.0" you're trying to use.
  2. If (1) looks fine, you can open a new issue at
     https://github.com/numpy/numpy/issues.  Please include details on:
     - how you installed Python
     - how you installed numpy
     - your operating system
     - whether or not you have multiple versions of Python installed
     - if you built from source, your compiler versions and ideally a build log

- If you're working with a numpy git repository, try `git clean -xdf`
  (removes all files not under version control) and rebuild numpy.

**Comment by seberg** [comment]
@marinaesp sorry that you are running into this. Can you check the updated help page at:

https://numpy.org/devdocs/user/troubleshooting-importerror.html

Since you are using a miniconda, I could imagine that this is an issue with activating the conda environment where NumPy is installed.

**Comment by marinaesp** [comment]
Thank you @seberg, i will check right now!

**Comment by seberg** [comment]
I am not sure how you are starting R, so I am a bit surprised it would not activate automatically.

Maybe you can try in a shell to run the conda activation manually and then `R`:
```
conda activate r-reticulate
R
#...
```

If `r-miniconda` or `r-reticulate` are packages you downloaded, you are likely better of asking them about it. They would have more experience with the setup.

**Comment by marinaesp** [comment]
> Maybe you can try in a shell to run the conda activation manually and then `R`:
> 
> ```
> conda activate r-reticulate
> R
> #...
> ```

Just tried that but no success. Maybe be uninstalling and installing python again may help?

**Comment by rossbar** [comment]
Can you try the following:
```
conda activate r-reticulate
python -c "import numpy as np"
```
If you get the error that you reported in the top comment, then that means something is likely wrong with the numpy installation / environment management.

If you don't see the error doing this, then this might be more related to how R/reticulate handles the environments etc., in which case you may have more luck on the reticulate issue tracker.

**Comment by seberg** [comment]
Looking at the initial report you filed. It does look like everything should be fine. You could check `PYTHONPATH` and `PATH` to be completely sure there is nothing there that may confuse things?  I am a bit surprised that you need the `reticulate::use_python` since you do activate the conda environment using `reticulate`, but I do not know that.

**Comment by marinaesp** [comment]
> Can you try the following:
> 
> ```
> conda activate r-reticulate
> python -c "import numpy as np"
> ```
> 
> If you get the error that you reported in the top comment, then that means something is likely wrong with the numpy installation / environment management.
> 
> If you don't see the error doing this, then this might be more related to how R/reticulate handles the environments etc., in which case you may have more luck on the reticulate issue tracker.

I get an error: "Could not find conda environment: r-reticulate".

I just updated the conda, and not the python configuration looks like this (if it may help):

`reticulate::py_config()`
python:         /Users/marinaespinasse/Library/r-miniconda/envs/r-reticulate/bin/python
libpython:      /Users/marinaespinasse/Library/r-miniconda/envs/r-reticulate/lib/libpython3.9.dylib[NOT FOUND]
pythonhome:     /Users/marinaespinasse/Library/r-miniconda/envs/r-reticulate:/Users/marinaespinasse/Library/r-miniconda/envs/r-reticulate
version:        3.9.0 | packaged by conda-forge | (default, Oct  9 2020, 10:34:32)  [Clang 10.0.1 ]
numpy:           [NOT FOUND]

Now it says that numpy is not found. I try to install it in the terminal, and see the there is a problem with solving the environment. What may it mean, if any of you know?

`(base) HI-11056:~ marinaespinasse$ conda install numpy
Collecting package metadata (current_repodata.json): done
Solving environment: failed with initial frozen solve. Retrying with flexible solve.
Solving environment: failed with repodata from current_repodata.json, will retry with next repodata source.
Collecting package metadata (repodata.json): done
Solving environment: |`

**Comment by seberg** [comment]
@marinaesp I don't have any more immediate ideas as to what to try.  It might be incorrect PATH variables or even an `LD_LIBRARY_PATH `, or failures to activate the conda environment correctly.  Maybe someone here will have an idea of what is going on, but you may have more luck asking at a reticulate or a conda help forum :(.

Unfortunately, NumPy is a typical symptom for these issues, which is why we try to compile common reasons and solutions, but it is hardly ever the underlying cause.  Once you/we find the solution, we should consider adding it to the troubleshooting website, so that it is less frustrating for the next person.

**Comment by marinaesp** [comment]
@seberg , yes looks like there is some issue with an environment, but this is beyond my knowledge.
Anyway, thank you for trying to help!

**Comment by marinaesp** [comment]
Hey @seberg and @rossbar,

I wanted to let you know that I managed to import NumPy into R. I restarted the mac and also updated the mac's Catalina OS to the latest version, and after that also updated the conda as
`conda update -n base -c defaults conda` in the Terminal. 

Then, importing NumPy using `reticulate` in R just worked! I cannot be sure, but I think it might have been a mac OS issue with conda, because my colleague on Windows PC could import NumPy with no problem. 
Maybe worth noting on the NumPy troubleshoot page for mac users that there might be issues with NumPy under older versions of Catalina OS.
